lib/auth/access.go

```diff
@@ -77,8 +85,45 @@ func (a *Server) DeleteRole(ctx context.Context, name string) error {
 				// Mask the actual error here as it could be used to enumerate users
 				// within the system.
 				log.Warnf("Failed to delete role: role %v is used by user cert authority %v", name, a.GetClusterName())
-				return trace.BadParameter("failed to delete a role that is still in use by a user, check the system server logs for more details")
+				return trace.Wrap(errDeleteRoleCA)
+			}
+		}
+	}
+
+	var nextToken string
+	for {
+		var accessLists []*accesslist.AccessList
+		var err error
+		accessLists, nextToken, err = a.Services.AccessListClient().ListAccessLists(ctx, 0 /* default page size */, nextToken)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+
+		for _, accessList := range accessLists {
+			for _, r := range accessList.Spec.Grants.Roles {
```

Comment:
Nit: I think we have a "contains" method for string slices somewhere in utils which will make this code a bit cleaner and avoid 3rd nested for loop.

Reply:
I've elected to use `slices` instead here, let me know what you think.

Reply:
Even better.

Reply:
Just dropping this here: https://github.com/gravitational/teleport/pull/33851 - this is the equivalent slices change on master that keeps master and branch/v14 in sync.